bench/postgres/optimized/mxactdesc.ll

```diff
@@ -31,74 +31,75 @@ define dso_local void @multixact_desc(ptr noundef %0, ptr nocapture noundef read
   %6 = load ptr, ptr %5, align 8
   %7 = getelementptr inbounds i8, ptr %4, i64 56
   %8 = load i8, ptr %7, align 8
-  %or.cond = icmp ult i8 %8, 32
-  br i1 %or.cond, label %9, label %10
+  %9 = and i8 %8, -16
+  %10 = and i8 %8, 16
+  %or.cond = icmp eq i8 %10, %9
```

Comment:
Regression.

Reply:
This is weird, we handle:
```
define i1 @foo(i8 %x) {
  %x16 = and i8 %x, 16
  %xm16 = and i8 %x, -16
  %r = icmp eq i8 %x16, %xm16
  ret i1 %r
}
```

as:
```
define i1 @foo(i8 %x) {
  %r = icmp ult i8 %x, 32
  ret i1 %r
}
```

directly. Not sure why this ending up unsimplified. @nikic any ideas?

Reply:
Looks like a multi-use problem: https://llvm.godbolt.org/z/Y9zbMdqYK

Reply:
Note that there's an intermediate transform to:
```llvm

define i1 @foo(i8 %x) {
  %1 = and i8 -32, %x
  %r = icmp eq i8 %1, 0
  ret i1 %r
}
```
before we get to the single icmp form.

Reply:
+1, I'll fix this up.